csharp/Schema/GroupNode.cs

```diff
@@ -37,11 +37,24 @@ public int FieldIndex(Node node)
 
         public override Node DeepClone()
         {
-            return new GroupNode(
-                Name,
-                Repetition,
-                Fields.Select(f => f.DeepClone()).ToArray(),
-                LogicalType is NoneLogicalType ? null : LogicalType);
+            using var logicalType = LogicalType;
+            var fields = Fields;
+            var clonedFields = fields.Select(f => f.DeepClone()).ToArray();
+            try
+            {
+                return new GroupNode(
+                    Name,
+                    Repetition,
+                    fields,
+                    logicalType is NoneLogicalType ? null : logicalType);
+            }
+            finally
+            {
+                foreach (var field in fields.Concat(clonedFields))
```

Comment:
Is it correct to dispose `clonedFields` here ?

Reply:
Yes, once we've created the GroupNode, it no longer references the original C# side fields but internally in the C++ GroupNode it has shared pointers that point to the fields. Disposing of the field just deletes the shared pointer to the C++ field, but the field itself won't be deleted due to the group node still holding a shared pointer to it.

Reply:
👍